src/Microsoft.Extensions.Primitives/StringSegment.cs

```diff
@@ -152,6 +184,21 @@ public bool Equals(StringSegment other, StringComparison comparisonType)
             return string.Compare(Buffer, Offset, other.Buffer, other.Offset, textLength, comparisonType) == 0;
         }
 
+        // This handles StringSegment.Equals(string, StringSegment, StringComparison) and StringSegment.Equals(StringSegment, string, StringComparison)
+        // vai the implicit type converter
```

Comment:
Couldn't we just add an instance method `Equals(string)` to this? The implicit operator has marginal utility outside of this given that we use `var` typing everywhere.